tests/test_dynamodb.py

```diff
@@ -80,6 +80,26 @@ def test_tag_filter(self):
         tag_map = {t["Key"]: t["Value"] for t in tags["Tags"]}
         self.assertTrue("test_key" in tag_map)
 
+    def test_kms_key_filter(self):
+        session_factory = self.replay_flight_data("test_dynamodb_kms_key_filter")
+        p = self.load_policy(
+            {
+                "name": "dynamodb-kms-key-filters",
+                "resource": "dynamodb-table",
+                "filters": [
+                    {
+                        "type": "kms-key",
+                        "key": "c7n:AliasName",
+                        "value": "^(alias/aws/dynamodb)",
+                        "op": "regex"
+                    }
+                ]
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
```

Comment:
should also add an assertion of the table name attribute to ensure its actually getting the intended resource, the cardinality checks are a weak form of verification that we've been moving away from in favor of an attribute check.

Reply:
Good point. Added a new assert on the name